hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -209,9 +224,16 @@ public IOException getIoException() {
   @Override
   public void write(int b) throws IOException {
     checkOpen();
-    byte[] buf = new byte[1];
-    buf[0] = (byte) b;
-    write(buf, 0, 1);
+    if (currentBufferRemaining == 0) {
+      allocateNewBuffer();
+    }
```

Comment:
For consistency, I think we should move the condition to `allocateNewBuffer` and rename it to `allocateNewBufferIfNeeded`.